src/main/java/org/jenkinsci/plugins/workflow/graphanalysis/ForkScanner.java

```diff
@@ -565,12 +569,11 @@ private static FlowNode findLastStartedNode(@Nonnull List<FlowNode> candidates)
             long startTime = Long.MIN_VALUE;
             for(FlowNode f : candidates) {
                 TimingAction ta = f.getAction(TimingAction.class);
-                if (ta != null) {
-                    long myStart = ta.getStartTime();
-                    if (myStart > startTime) {
-                        returnOut = f;
-                        startTime = myStart;
-                    }
+                // Null timing with multiple heads is probably a node where the GraphListener hasn't fired to add TimingAction yet
```

Comment:
See also https://github.com/jenkinsci/workflow-support-plugin/pull/16 which fixes the SemaphoreStep issue from [JENKINS-38737](https://issues.jenkins-ci.org/browse/JENKINS-38737)